lib/elixir/lib/system.ex

```diff
@@ -5,40 +5,71 @@ defmodule System do
   with the VM or the host system.
   """
 
-  defp strip_re(iodata, pattern) do
-    :re.replace(iodata, pattern, "", [return: :binary])
+  @base_dir           Path.join([__DIR__, "..", "..", ".."])
+  @version_file       Path.join(@base_dir, "VERSION")
+  @git_dir            Path.join(@base_dir, ".git")
+  @git_head_file      Path.join(@git_dir, "HEAD")
+
+  @spec strip(iodata) :: String.t
+  defp strip(iodata) do
+    iodata
+    |> :re.replace("^[\s\r\n\t]+", "", [return: :binary])
+    |> :re.replace("[\s\r\n\t]+$", "", [return: :binary])
```

Comment:
@lexmag and @josevalim: since strip was used in other files, and after different trials, I ended up with an improved version of the original function
```
:re.replace(iodata, "^[\s\r\n\t]+|[\s\r\n\t]+$", "", [:global, return: :binary])
```